core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -71,6 +72,16 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val actorSyste
   // Executors we have requested the cluster manager to kill that have not died yet
   private val executorsPendingToRemove = new HashSet[String]
 
+  /**
+   * Send new credentials to executors. This is the method that is called when the scheduled
+   * login completes, so the new credentials can be sent to the executors.
+   * @param credentials
+   */
+  def sendNewCredentialsToExecutors(credentials: SerializableBuffer): Unit = {
+    // We don't care about the reply, so going to deadLetters is fine.
```

Comment:
Ah, I just realized we'd have to have executors ask for new credentials - else we will not be resilient to executor failures. To ensure that once new executors start up, they get the latest tokens, we have to make sure they ask for the new tokens from the AM